src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Compiler/AddConstraintValidatorsPass.php

```diff
@@ -26,22 +27,13 @@ public function process(ContainerBuilder $container)
         $validators = array();
         foreach ($container->findTaggedServiceIds('validator.constraint_validator') as $id => $attributes) {
             if (isset($attributes[0]['alias'])) {
-                $validators[$attributes[0]['alias']] = $id;
+                $validators[$attributes[0]['alias']] = new Reference($id);
             }
 
             $definition = $container->getDefinition($id);
-
-            if (!$definition->isPublic()) {
-                throw new InvalidArgumentException(sprintf('The service "%s" must be public as it can be lazy-loaded.', $id));
-            }
-
-            if ($definition->isAbstract()) {
```

Comment:
Abstract services should be ignored ie you should "continue". Note that this applies to many other cases in the code base: many passes throw on abstract services, but now that tags can be inherited in child definitions, we should just "continue" insteand. Deserves another PR if you'd like to do it :)